### PR TITLE
Feature/zoom

### LINKED ISSUE #302: Zoom not behaving as expected
If you scroll to much you run out of canvas.

### LINKED ISSUE #343: Arrow in canvas table is too big

### PR DESCRIPTION
**Pull request by deletidev**
Closed #302 
Closed #343

### CODE REVIEW

File: src/pods/toolbar/components/zoom-button/zoom-in-button.component.tsx [hunk 1]
```diff
@@ -4,14 +4,15 @@ import { ZoomIn } from '@/common/components/icons';
...
-  const { zoomIn } = useCanvasViewSettingsContext();
+  const { zoomIn, canvasViewSettings } = useCanvasViewSettingsContext();
 
   return (
     <ToolbarButton
       icon={<ZoomIn />}
       label="Zoom In"
       onClick={zoomIn}
       className={classes.button}
+      disabled={canvasViewSettings.zoomFactor < 4.6}
```
Comment (brauliodiez) [inline src/pods/toolbar/components/zoom-button/zoom-in-button.component.tsx:15]: Move this to a const, so it's easier to find and has more meaning, avoid magic numbers.

```ts
const MINIMUM_ZOOM_FACTOR_ALLOWED = 4.6
```

File: src/pods/toolbar/components/zoom-button/zoom-out-button.component.tsx [hunk 2]
```diff
@@ -4,14 +4,15 @@ import { ZoomOut } from '@/common/components/icons/zoom-out-icon.component';
...
-  const { zoomOut } = useCanvasViewSettingsContext();
+  const { zoomOut, canvasViewSettings } = useCanvasViewSettingsContext();
 
   return (
     <ToolbarButton
       icon={<ZoomOut />}
       label="Zoom Out"
       onClick={zoomOut}
       className={classes.button}
+      disabled={canvasViewSettings.zoomFactor > 15.01}
```
Comment (brauliodiez) [inline src/pods/toolbar/components/zoom-button/zoom-out-button.component.tsx:15]: Same here

```ts
const MAXIMUM_ZOOM_FACTOR_ALLOWED = 15.01;
```

File: src/pods/canvas/components/table/components/database-table-row.component.tsx [hunk 3]
```diff
@@ -31,7 +31,8 @@ const COLUMN_NAME_PIXEL_WIDTH = calculateColumNameWidth(
   TABLE_CLEAN_WIDTH
 );
 
-const SPACE_ARROW_TEXT = TABLE_CONST.FONT_SIZE * 1.5;
+const SPACE_ARROW_TEXT = TABLE_CONST.FONT_SIZE * 1.35;
+const PIXELS_CENTER_ARROW = 1.5;
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/components/database-table-row.component.tsx:35]: I guess this tweakings are not related with Zoom issue? If that's the case for the next time, create separate issue and branch for this, right now is adding noise to the PR (if not please could you indicate how is related to this zoom factor)?

File: src/pods/canvas/canvas-svg.component.tsx [hunk 4]
```diff
@@ -41,8 +41,8 @@ export const CanvasSvgComponent: React.FC<Props> = props => {
       xmlns="http://www.w3.org/2000/svg"
       className={classes.containerSvg}
       viewBox={`0 0 ${viewBoxSize.width} ${viewBoxSize.height}`}
-      width={canvasSize.width}
-      height={canvasSize.height}
+      width={20000}
+      height={20000}
```
Comment (brauliodiez) [inline src/pods/canvas/canvas-svg.component.tsx:45]: Move this to const

```ts
const CANVAS_WIDTH = 20000;
const CANVAS_HEIGHT = 20000;
```

Comment (deletidev) [reply src/pods/canvas/canvas-svg.component.tsx]: It's fixed.

File: src/pods/canvas/components/table/components/database-table-row.component.tsx (hunk 3 above)
Comment (deletidev) [reply src/pods/canvas/components/table/components/database-table-row.component.tsx:35]: Sorry, I noticed that mistake while playing with the zoom, next time I will create a new issue.

File: src/core/providers/canvas-view-settings/canvas-view-settings.model.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ export interface CanvasViewSettingsModel {
 
 export const createInitialSettings = () => ({
   canvasSize: { width: 2400, height: 2400 },
-  zoomFactor: 1,
+  zoomFactor: 8.5,
```
Comment (brauliodiez) [inline src/core/providers/canvas-view-settings/canvas-view-settings.model.ts:13]: Move it const in the same file:

```diff
+ CONST CANVAS_SIZE = { width: 2400, height: 2400 };
+ const DEFAULT_ZOOM_FACTOR = 8.5;

export const createInitialSettings = () => ({
-  canvasSize: { width: 2400, height: 2400 },
- canvasSize: CANVAS_SIZE,
-  zoomFactor: 8.5,
+  zoomFactor: DEFAULT_ZOOM_FACTOR,
```

File: src/pods/toolbar/components/zoom-button/zoom-out-button.component.tsx (hunk 2 above)
Comment (deletidev) [reply src/pods/toolbar/components/zoom-button/zoom-out-button.component.tsx]: It's fixed

File: src/pods/toolbar/components/zoom-button/zoom-in-button.component.tsx (hunk 1 above)
Comment (deletidev) [reply src/pods/toolbar/components/zoom-button/zoom-in-button.component.tsx]: It's fixed

File: src/core/providers/canvas-view-settings/canvas-view-settings.model.ts (hunk 5 above)
Comment (deletidev) [reply src/core/providers/canvas-view-settings/canvas-view-settings.model.ts]: It's fixed
